File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Type def - Interface 01.fs
module Module

type IInherit1 =
    inherit IInterface

type IInherit2 =
    interface IInterface


type EmptyType() =
    interface IEmptyInterface

let e = EmptyType()
e :> IEmptyInterface |> ignore
e :> IEmptyBaseInterface |> ignore


type Type() =
    interface IInterface with
        member this.M1() = ()
        member this.M2(i: int) = ()

let t = Type()
t :> IInterface |> ignore
t :> IBaseInterface |> ignore

File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Type def - Interface 01.cs
public interface IBaseInterface
{
  void M1();
  int P1 { get; }
}


public interface IInterface : IBaseInterface
{
  void M2(int i);
  int P2 { get; }
}


public interface IEmptyBaseInterface
{
}

public interface IEmptyInterface : IEmptyBaseInterface
{
}
